packages/http-client-csharp/emitter/src/emitter.ts

```diff
@@ -180,6 +199,63 @@ export async function $onEmit(context: EmitContext<NetEmitterOptions>) {
   }
 }
 
+/** check the dotnet sdk installation.
+ * Report diagnostic if dotnet sdk is not installed or its version does not meet prerequisite
+ * @param program The typespec compiler program
+ * @param minVersionRequisite The minimum required major version
+ */
+async function validateDotNetSdk(program: Program, minMajorVersion: number): Promise<boolean> {
+  try {
+    const result = await execAsync("dotnet", ["--version"], { stdio: "pipe" });
+    return validateDotNetSdkVersion(program, result.stdout, minMajorVersion);
+  } catch (error: any) {
+    if (error && "code" in (error as {}) && error["code"] === "ENOENT") {
+      reportDiagnostic(program, {
+        code: "invalid-dotnet-sdk-dependency",
+        messageId: "missing",
+        format: {
+          dotnetMajorVersion: `${minMajorVersion}`,
+          downloadUrl: "https://dotnet.microsoft.com/",
+        },
+        target: NoTarget,
+      });
+    }
+    return false;
+  }
+}
+
+export function validateDotNetSdkVersion(
+  program: Program,
+  version: string,
+  minMajorVersion: number,
+): boolean {
+  if (version) {
+    const versions = version.split(".");
+    if (versions.length < 3) {
+      Logger.getInstance().error("Invalid .NET SDK version.");
+      return false;
+    }
+    const major = +versions[0];
```

Comment:
```suggestion
    const dotIndex = version.indexOf('.');
    const firstPart = dotIndex === -1 ? version : version.substring(0, dotIndex);
    const major = Number(firstPart);
    
    if (isNaN(major)) {
      Logger.getInstance().error("Invalid .NET SDK version.");
      return false;
    }   
```

Since we're only validating the major, there's no benefit to making assumptions about the entire version pattern - we only care that the first part is numeric and in the range that we care about.  There's no reason to parse out the entire string nor allocate an array.